experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -234,6 +230,16 @@ export type ContextuallyTypedNodeData =
  */
 export type ContextuallyTypedFieldData = ContextuallyTypedNodeData | undefined;
 
+/**
+ * context object providing the typeSet and SchemaData and optional globalFieldKeySymbol to provide context to the data
+ * @alpha
+ */
+export interface ContextObject {
```

Comment:
Due to how our exporting works, package exported names arn't namespaces under the module they are part of. This means types should be named so they are sufficiently descriptive relative to everything else at the package level.

Given that basically all interfaces are for "objects", that part of the name adds almost nothing, and we already have lots of things with "context" thats also not very descriptive.

I think we either need to avoid exporting this from the package (and make it an implementation detail) or get it a more descriptive name.

Reply:
Replaced `ContextObject` with new interface `TreeDataContext` suggested below